cp23/hpi-fun2/cubes/executables/generator.cpp

```diff
@@ -30,12 +45,52 @@ void sample(int num, string_view content)
     predefined("sample" + num_str, "Sample #" + num_str, content);
 }
 
+vector<array<long long, 3>> generate_coords(int width, int length, int height)
+{
+    vector<array<long long, 3>> coords;
+
+    const long long base_x_coordinate = rnd.next(MAX_COORDINATE);
+    const long long base_y_coordinate = rnd.next(MAX_COORDINATE);
+    const long long base_z_coordinate = rnd.next(MAX_COORDINATE);
+
+    // Threshold for the cloud block
+    const float THRESHOLD = 0.4;
+
+    FastNoiseLite noiseGenerator;
+
+    // Configure the noise
+    noiseGenerator.SetNoiseType(FastNoiseLite::NoiseType_ValueCubic);
+    noiseGenerator.SetFrequency(0.1);
```

Comment:
You may want to alter the thresholds a bit. One more dense, one more sparse random testcase.